packages/bindings/src/query.rs

```diff
@@ -25,7 +26,7 @@ pub enum OsmosisQuery {
     EstimatePrice {
         first: Swap,
         route: Vec<Step>,
-        amount: SwapAmount,
+        amount: SwapAmountWithLimit,
```

Comment:
The actual query structure in the gamm module uses SwapAmount.

The querier than sets a min of 1 (or max of huge number) before running it. Let's keep that behavior. The estimate should return a value and not error.

Reply:
However, you do rightly mention the need for a `sender` field.

Looking at the code, I see they actually run the transaction (on some scratch data they don't write to disk). You need to run it with a sender address that has enough tokens to cover this estimation.

Basically, it is a dry run of a given account and returns what would have happened.
Generally a contract can provide it's own address here if it is preparing for such a swap.
This is a bit annoying that you cannot really estimate arbitrary swaps (I think... need to review that more)

Reply:
> The querier than sets a min of 1 (or max of huge number) before running it. Let's keep that behavior. The estimate should return a value and not error.

I did something like that in the impl, except with zero instead of one. Now, this matched so nicely with the `WithLimit` that I thought that was the right thing to do. Will revert.